service/storage_proxy.hh

```diff
@@ -584,6 +584,9 @@ public:
     void allow_replaying_hints() noexcept;
     future<> drain_on_shutdown();
 
+    void set_hints_host_filter(const db::hints::host_filter& new_filter);
+    db::hints::host_filter get_hints_host_filter() const;
+
     const stats& get_stats() const {
```

Comment:
There is a problem with two configuration sources. Suppose you set via REST and then SIGHUP for some unrelated change. Which should take precedence? I think REST will now not be overridden by SIGHUP, but it's really an implementation accident.

We should route REST via config, not have two separate paths talking to the controlled object.

I think this is a preexisting problem.

Reply:
In the case you described, SIGHUP won't cause observers to be notified, because the value that was re-read from config compares equal with the value that was parsed at startup (or at the previous reload). And yes, it is clearly an accident.

I'm not sure which way should have precedence, or if there should be precedence at all. I know that, for example, CLI arguments have precedence over configuration, so that if you specify `--hinted-handoff-enabled` in CLI, SIGHUP-reloading from config won't work for it.

An idea: maybe we should have a generic API for live-updating configuration parameters through REST? It would override CLI and YAML configuration when set, but could be cleared so that the CLI or config parameter is available again.

> We should route REST via config, not have two separate paths talking to the controlled object.

Could you clarify what you meant? Does it mean that REST should modify the `config` object directly so that SIGHUP in the scenario above would cause the host_filter to be updated?

Reply:
Too much time has passed, so I don't know if that's what I meant. It some seem reasonable to consolidate REST config changes into db::config. There is still the problem that there can be confusion between static (yaml) and dynamic (REST) configuration, but at least db::config/named_value can keep track of it (new config_source enum).

Reply:
The v3 avoids this problem for now by allowing to change the config through HTTP API only.